### ISSUE TITLE
Deprecate tasks.findByPath

### ISSUE THREAD

**Issue by avonengel** [gradle/gradle#1583]
### Expected Behavior
As the publishing extension is not accessed by `project.tasks.findByPath(":tasks")` (at least from a user's perspective), the publishing extension should not get "locked".

### Current Behavior
For some reason, calling `tasks.findByPath` causes the publishing extension to be accessed. This leads to subsequent build-failure when the publishing extension is configured.

### Context
We used `findByPath` to be able to correctly find tasks in a multi-project build. It would be convenient, if you could provide a workaround that does not cause the publishing extension to get write-locked.

Also see #1470 

### Steps to Reproduce (for bugs)

Minimal example:
```
plugins {
    id 'maven-publish'
}

project.tasks.findByPath(":tasks")

publishing {
    publications {
        maven(MavenPublication) {}
    }
}
```
Simply run `gradle tasks`.

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Temp\maven-publish-bug\build.gradle' line: 7

* What went wrong:
A problem occurred evaluating root project 'maven-publish-bug'.
> Cannot configure the 'publishing' extension after it has been accessed.
```

### Your Environment

 * Build scan URL: N/A, as the build configuration fails
* Gradle version: tested with 3.4.1, 2.14.1

**Comment by oehme** [comment]
We should deprecate and remove `tasks.findByPath`, because it creates confusing behavior. Please use `findByName` instead. Your example doesn't show why you are using it, but there is always a better way than `findByPath`. If you elaborate on your use case, we can find a solution.

**Comment by avonengel** [comment]
I do not remember why we used `findByPath`, but it had something to do with multi-project builds. I worked around this by using `findByName`, as you also suggested. Seems to be solved, as I didn't hear of any further problems with this since then (shortly after I created this ticket).

I don't have an immediate problem with this, but it probably remains unsolved - I didn't try again with Gradle 4.1.
